community/modules/scripts/spack-setup/templates/spack_setup.yml.tftpl

```diff
@@ -127,9 +133,24 @@
     when: lock_out.rc == 0
 
   - name: Wait for lock
-    ansible.builtin.wait_for:
-      path: "{{ lock_dir }}/done"
-      state: present
-      timeout: 600
-      sleep: 10
-    when: lock_out.rc != 0
+    block: 
+    - name: Wait for lock
+      ansible.builtin.wait_for:
+        path: "{{ lock_dir }}/done"
+        state: present
+        timeout: 600
+        sleep: 10
+      when: lock_out.rc != 0
+
+    rescue:
+    - name: Timed out on lock, get install directory contents
+      ansible.builtin.command: "ls -ltra {{ lock_dir }}"
```

Comment:
I think the [find](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/find_module.html) module is the ansible way to do this.

Reply:
Done